xml-occurrence-parsers/src/main/java/org/gbif/xml/occurrence/parser/identifier/OccurrenceKeyHelper.java

```diff
@@ -19,34 +23,30 @@ public static String buildKeyPrefix(String datasetKey) {
 
   @Nullable
   public static String buildKey(@Nullable HolyTriplet triplet) {
-    if (triplet == null
-        || triplet.getDatasetKey() == null
-        || triplet.getInstitutionCode() == null
-        || triplet.getCollectionCode() == null
-        || triplet.getCatalogNumber() == null) {
+    if (Objects.isNull(triplet)
```

Comment:
this is repeated in L63, should we create a method private boolen isHolyTriplet(HolyTriplet) ?

Reply:
I suggest refactoring some old parts in a separated task